unity-client/Assets/Scripts/MainScripts/DCL/Components/Animator/DCLAnimator.cs

```diff
@@ -33,9 +34,23 @@ public DCLAnimationState Clone()
             }
 
             public DCLAnimationState[] states;
-        }
 
-        public Model model = new Model();
+            public override BaseModel GetDataFromJSON(string json)
+            {
+                return Utils.SafeFromJson<Model>(json);
+            }
+
+            public override bool Equals(object obj)
+            {
+                return obj is Model model &&
+                       EqualityComparer<DCLAnimationState[]>.Default.Equals(states, model.states);
+            }
+
+            public override int GetHashCode()
+            {
+                return 282409271 + EqualityComparer<DCLAnimationState[]>.Default.GetHashCode(states);
```

Comment:
what is this magic number?

Reply:
This magic number is a random constant, The function GetHashCode is used to generate a unique int identifier for Dictionaries keys (this way you only compare an int instead of the whole objects, that is the reason why dictionaries are more  efficient)

So for each class you need a random intenger to start the hash to be sure that you are comparing a correct key so you don't have problem with parent/children..etcs

Reply:
How are we preventing collisions?

Reply:
@Suduck can we create a static class with all those magic numbers for the hashing as constants?